[CI] CorruptedFileIT testReplicaCorruption failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/ihixbkyjmor5k/tests/:server:internalClusterTest/org.elasticsearch.index.store.CorruptedFileIT/testReplicaCorruption


**Reproduction line:**
```
./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.index.store.CorruptedFileIT.testReplicaCorruption" -Dtests.seed=1111C7E1A6B1F7DB -Dtests.locale=ru -Dtests.timezone=Asia/Kuala_Lumpur -Druntime.java=21
```

**Applicable branches:**
main

**Reproduces locally?:**
No

**Failure history:**
[Failure dashboard for `org.elasticsearch.index.store.CorruptedFileIT#testReplicaCorruption`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testReplicaCorruption'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.index.store.CorruptedFileIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))


**Failure excerpt:**
```
java.lang.AssertionError: Unexpected AlreadyClosedException

  at __randomizedtesting.SeedInfo.seed([1111C7E1A6B1F7DB]:0)
  at org.elasticsearch.index.engine.InternalEngine.failOnTragicEvent(InternalEngine.java:2523)
  at org.elasticsearch.index.engine.InternalEngine.refresh(InternalEngine.java:2069)
  at org.elasticsearch.index.engine.InternalEngine.refresh(InternalEngine.java:2016)
  at org.elasticsearch.index.shard.IndexShard.resetEngineToGlobalCheckpoint(IndexShard.java:4229)
  at org.elasticsearch.index.shard.IndexShard.lambda$innerAcquireReplicaOperationPermit$45(IndexShard.java:3744)
  at org.elasticsearch.index.shard.IndexShard$6.onResponse(IndexShard.java:3599)
  at org.elasticsearch.index.shard.IndexShard$6.onResponse(IndexShard.java:3569)
  at org.elasticsearch.index.shard.IndexShard.lambda$asyncBlockOperations$39(IndexShard.java:3526)
  at org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:171)
  at org.elasticsearch.action.ActionListener$3.onResponse(ActionListener.java:314)
  at org.elasticsearch.index.shard.IndexShardOperationPermits$1.doRun(IndexShardOperationPermits.java:118)
  at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)
  at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
  at java.lang.Thread.run(Thread.java:1583)

  Caused by: org.apache.lucene.store.AlreadyClosedException: engine is closed

    at org.elasticsearch.index.shard.IndexShard.getEngine(IndexShard.java:3184)
    at org.elasticsearch.index.shard.IndexShard$RefreshFieldHasValueListener.afterRefresh(IndexShard.java:4014)
    at org.apache.lucene.search.ReferenceManager.notifyRefreshListenersRefreshed(ReferenceManager.java:275)
    at org.apache.lucene.search.ReferenceManager.doMaybeRefresh(ReferenceManager.java:182)
    at org.apache.lucene.search.ReferenceManager.maybeRefreshBlocking(ReferenceManager.java:240)
    at org.elasticsearch.index.engine.InternalEngine.refresh(InternalEngine.java:2045)
    at org.elasticsearch.index.engine.InternalEngine.refresh(InternalEngine.java:2016)
    at org.elasticsearch.index.shard.IndexShard.resetEngineToGlobalCheckpoint(IndexShard.java:4229)
    at org.elasticsearch.index.shard.IndexShard.lambda$innerAcquireReplicaOperationPermit$45(IndexShard.java:3744)
    at org.elasticsearch.index.shard.IndexShard$6.onResponse(IndexShard.java:3599)
    at org.elasticsearch.index.shard.IndexShard$6.onResponse(IndexShard.java:3569)
    at org.elasticsearch.index.shard.IndexShard.lambda$asyncBlockOperations$39(IndexShard.java:3526)
    at org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:171)
    at org.elasticsearch.action.ActionListener$3.onResponse(ActionListener.java:314)
    at org.elasticsearch.index.shard.IndexShardOperationPermits$1.doRun(IndexShardOperationPermits.java:118)
    at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)
    at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
    at java.lang.Thread.run(Thread.java:1583)

```

--- a/server/src/main/java/org/elasticsearch/index/shard/IndexShard.java
+++ b/server/src/main/java/org/elasticsearch/index/shard/IndexShard.java
@@ -4013,6 +4013,8 @@ public class IndexShard extends AbstractIndexShardComponent implements IndicesCl
             if (enableFieldHasValue) {
                 try (Engine.Searcher hasValueSearcher = getEngine().acquireSearcher("field_has_value")) {
                     setFieldInfos(FieldInfos.getMergedFieldInfos(hasValueSearcher.getIndexReader()));
+                } catch (AlreadyClosedException ignore) {
+                    // engine is closed - no updated FieldInfos is fine
                 }
             }
         }

